[PLT-7396] Add the ability to revoke user sessions in System Console > Users #7493

rickbatka opened the pull request:
#### NOTE! This PR is part of a group of PRs with identical title in the following 4 repos, all of which must be merged simultaneously:
mattermost-server
mattermost-redux
mattermost-webapp
mattermost-api-reference

#### Summary
Adds api endpoint for revoking all sessions for a user, as used by this new feature accessed from the system users dropdown.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/7493

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs)
- [x] All new/modified APIs include changes to the drivers
- [ ] Has enterprise changes (please link)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json) updates
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

jasonblais commented:
Thanks @rickbatka, looks like there's a build failure, but otherwise this one is ready to merge

jwilander commented:
@rickbatka could give this a rebase? Not sure why the tests are failing where they are but they're passing on master

rickbatka commented:
Ok, sorry but I'm new to all this - you're going to have to be more specific with what you need. I really don't know how to interpret that build failure. I think this is the relevant info from the build logs, but I don't know what to make of it  (it doesn't look to be related to anything I touched):

> make: *** [test-te] Error 2
> [TemplateProject] FAILED performing builders from: 'mattermost-platform-template-split'
> Build step 'Use builders from another project' marked build as failure

As for the rebase, what exactly do you want me to do? Fresh checkout from upstream,  rebase my changes as a single commit, and submit an entirely new PR? Should I close this one out then?

jwilander commented:
Ah you submitted the PR from your master branch, so that makes rebasing a little more difficult. Let me look into the build failure, I don't think it has anything to do with your code

ccbrown commented:
I don't actually see a cause for that error in the build log at all. Gonna restart it and see what happens.

ccbrown commented on api4/user_test.go:
```diff
@@ -1971,6 +1971,54 @@ func TestRevokeSessions(t *testing.T) {
 
+func TestRevokeAllSessions(t *testing.T) {
+	th := Setup().InitBasic()
+	defer TearDown()
```
> api4/user_test.go:1976:8: undefined: TearDown

Should be `defer th.TearDown()`.

rickbatka commented:
You're right; I don't know how I managed that. I must have been running a different test when I was developing, but I swear I started with it failing and worked on it until it was passing. I must be going crazy.

Anyway, I fixed that build breaker and the test was still failing, so I fixed that too. Sorry for the back-and-forth, hopefully I got it right this time (commit added to PR).

ccbrown commented:
Awesome. Everything looks good to me, and we can go ahead and merge this one before the others. Thanks!
